### PR TITLE
follower read: add new option for tidb_replica_read

### PR DESCRIPTION
**Pull request by hicqu**
### What is changed, added or deleted? (Required)

Add an new option for tidb_replica_read.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

https://github.com/pingcap/tidb/pull/15721

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: reference/performance/follower-read.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ This variable is used to set the data read mode expected by the current session.
 
 - When the value of `tidb_replica_read` is set to `leader` or an empty string, TiDB maintains its original behavior and sends all read operations to the leader replica to perform.
 - When the value of `tidb_replica_read` is set to `follower`, TiDB selects a follower replica of the Region to perform all read operations.
+- When the value of `tidb_replica_read` is set to `leader-and-follower`, TiDB can select any replicas to perform read operations.
```
Comment (zhangjinpeng1987) [inline reference/performance/follower-read.md:37]: Not include learner replica

Comment (hicqu) [reply reference/performance/follower-read.md:37]: Learners are included. In fact `follower` also includes them.

Comment (zhangjinpeng1987) [reply reference/performance/follower-read.md:37]: leader-and-follower also can read learner?

Comment (hicqu) [reply reference/performance/follower-read.md:37]: Both `leader-and-follower` and `follower` can read learner.

**Comment by TomShawn** [comment]
@hicqu Please specify all the affected document versions, thanks!

File: reference/performance/follower-read.md (hunk 1 above)
Comment (zhangjinpeng1987) [reply reference/performance/follower-read.md:37]: How about if a learner doesn't have enough data? For example, the replica is in balancing status.

Comment (hicqu) [reply reference/performance/follower-read.md:37]: If it's applying snapshots, new proposals will be dropped and TiDB clients will switch to a next peer.

Comment (shenli) [reply reference/performance/follower-read.md:37]: can we optimize it latter?  For example, peer could report its status to pd, and TiDB will not choose the peer who is applying a snapshot (or not ready).

Comment (hicqu) [reply reference/performance/follower-read.md:37]: Good idea. Seems it's possible.

**Pull request merged**